### ISSUE TITLE
Midpoint of -Inf, Inf being NaN is kinda useless. Why not have it be zero?

### ISSUE THREAD

**Issue by freemin7** [JuliaMath/Roots.jl#270]
It doesn't appear particularly useful to me that the midpoint of interval [-Inf,Inf] should be NaN for bisection. If the goal is to find a number such that both resulting intervals have the same number of floats the obvious result is zero. A patch to `__middle` would look like this 
```
if (0 == xor(reinterpret(S,typemax(T)),reinterpret(S,x)) & typemax(S)>>1)  && (0 ==(xor(reinterpret(S,typemax(T)),reinterpret(S,y)) & typemax(S)>>1)) return zero(T) end;
```
Or similar.

**Comment by jverzani** [comment]
This is right, but this case gets caught in `_middle` allowing `__middle` to be called without a branch. (The first one is only called in `init_state`, the latter each iteration, so presumably there is a performance benefit, though this method is about robustness more so than performance.)

**Comment by jverzani** [comment]
Closed by #272. Reopen if this isn't sufficient.
